oolab/src/main/java/agh/ics/oop/AbstractWorldMap.java

```diff
@@ -26,14 +30,9 @@ public boolean isOccupied(Vector2d position) {
 
     @Override
     public Object objectAt(Vector2d position) {
-        for (Animal zwierzatko : animals) {
-            if (zwierzatko.isAt(position)){
-                return zwierzatko;
-            }
-        }
-        for (Grass kepka : grasses) {
-            if (kepka.getPosition().equals(position)){
-                return kepka;
+        for (Vector2d animalPos : animals.keySet()) {
```

Comment:
tu nie wykorzystujemy potencjału mapy - po to ją mamy żeby nie przeszukiwać zbioru kluczy tylko użyć `get()` ;)